deployment/cloudbuild-deploy.yaml

```diff
@@ -0,0 +1,140 @@
+---
+steps:
+  # Build the container image
+  - name: 'gcr.io/cloud-builders/docker'
+    entrypoint: '/bin/bash'
+    args:
+      - '-c'
+      - |
+        # Write namespace to file for future usage
+        case "$BRANCH_NAME" in
+          master)
+            echo "dev" > .namespace
+            ;;
+          staging)
+            echo "staging" > .namespace
+            ;;
+          stable)
+            echo "prod" > .namespace
+            ;;
+        esac
+
+        # Build container
+        docker build \
+          -f deployment/dockerfiles/$(cat .namespace)/${_CONTAINER} \
```

Comment:
For my understanding, I assume the triggers will be in charge of determining what container(s) should be built depending on what files got changed in a PR ?

Reply:
Yea that is right, so I'll define a trigger for something like changes on `origin-dapp/src/*` and then give it the _CONTAINER variable `origin-dapp`.